### PR TITLE
Message and Draft objects expose their related events

### PR DESCRIPTION
**Pull request by zspencer**
Resolves https://github.com/nylas/nylas-ruby/issues/119

### CODE REVIEW

File: lib/restful_model_collection.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 module Nylas
   class RestfulModelCollection
 
-    attr_accessor :filters
+    attr_accessor :filters, :_api, :model_class
```
Comment (zspencer) [inline lib/restful_model_collection.rb]: I found this useful for debugging purposes, but I think I should probably revert this or mark them as private explicitly?

File: lib/message.rb [hunk 2]
```diff
@@ -71,8 +71,31 @@ def as_json(options = {})
...
+    def events?
+      !events.nil? && !events.empty?
+    end
+
+    def events=(events)
+      unless events.respond_to?(:map)
+        raise TypeError, "unable to iterate over #{events}, events must respond to #map"
+      end
+
+      @events = events.map do |event_data|
```
Comment (khamidou) [inline lib/message.rb]: Ah, I should have mentioned this – the event field is read-only so we don't need to be able to assign objects fields to it.

**Comment by khamidou** [comment]
Zee,

Thanks for the PR. I just realized reading it I didn't mention that the events field on a message is basically read-only (it exposes the events attached to a message). As such, we don't need to be able to assign individual events to it. Apart from this the PR LGTM.

File: lib/message.rb (hunk 2 above)
Comment (zspencer) [reply lib/message.rb]: Ah! good to know :)

File: lib/restful_model_collection.rb (hunk 1 above)
Comment (khamidou) [reply lib/restful_model_collection.rb]: Curious, could you expand on how this is useful for debugging?

Comment (zspencer) [reply lib/restful_model_collection.rb]: Sure! I do a lot of REPL driven exploration of the code, and in a few different cases as I was exploring the API I wanted to inject a different version of `_api` and/or `model_class` to get better insights into what was going on. Having these available allowed me to replace them without reaching in and changing the instance variable.

**Comment by zspencer** [comment]
@khamidou - OK this is ready for re-review, repositioned the interface to emphasize that events are read-only.

**Comment by khamidou** [comment]
LGTM, but should we add a short example in the README to show the new APIs?

**Comment by zspencer** [comment]
Added example to the messages section of the README. I would like to reinvest the README documentation into an interactive ruby file that people can use to get familiar with the API instead of a document. 

Thoughts on that in a later ticket?

**Pull request merged**